### ISSUE TITLE
effector-react not working with ink (component update)

### ISSUE THREAD

**Issue by darky** [effector/effector#773]
**What is the current behavior:**

Component update for Ink React component not working, when store is updated

**Please provide the steps to reproduce and if possible a minimal demo of the problem via https://share.effector.dev, https://codesandbox.io or similar**

```tsx
import * as React from 'react'

import { useStore } from 'effector-react'
import { render, Text } from 'ink'
import TextInput from 'ink-text-input'
import { createEvent, createStore } from 'effector'

const $testStore = createStore('')
const testEvent = createEvent<string>()
$testStore.on(testEvent, (_, v) => v)

export const Component = () => {
  const value = useStore($testStore)

  return (
    <>
      <TextInput value={value} onChange={(v) => testEvent(v)} />
      <Text>{value}</Text>
    </>
  )
}

render(<Component />)
```

**What is the expected behavior:**

When you type something in `TextInput`, updates of React `Component` expected.
But nothing happen. It's can only rendered once.
On Effector side all fine. `testEvent` triggered, `$testStore` state updated

**Which versions of effector packages, and which browser and OS are affected by this issue? Did this work in previous versions of effector?:**

effector: 22.3.0
effector-react: 22.1.6
ink: 3.2.0
ink-text-input: 4.0.3
react: 17.0.2

---

node.js: 18.8.0

**Comment by darky** [comment]
effector-react: 21.3.3
works fine

**Comment by AlexandrHoroshih** [comment]
Yeah, looks like there is a some issue with latest `effector-react`, which now uses official `use-sync-external-store` shim for React 17, and `ink` 🤔 

I suggest to keep using `effector-react@22.0.6` (latest version before `react@18` support) until `ink` is updated to React 18

**Comment by darky** [comment]
`22.0.6` not working too
`21.3.3` is latest working version

**Comment by zerobias** [comment]
`ink` should be maintained in order to work with newer react versions
